lib/cli/kit/command_registry.rb

```diff
@@ -0,0 +1,79 @@
+require 'cli/kit'
+
+module CLI
+  module Kit
+    module CommandRegistry
+      attr_accessor :commands, :aliases
+      class << self
+        attr_accessor :registry_target
+      end
+
+      def resolve_contextual_command
+        nil
+      end
+
+      def contextual_aliases
+        {}
+      end
+
+      def contextual_command_class(_name)
+        raise NotImplementedError
+      end
+
+      def self.extended(base)
+        raise "multiple registries unsupported" if registry_target
```

Comment:
`if self.registry_target` to be consistent with the next line?

Reply:
`self` is only needed on the next line because it's a setter, and otherwise it would create a variable called `registry_target` and set it to base.  `self.` convinces ruby that it's a method call.